libraries/chain/include/graphene/chain/protocol/event.hpp

```diff
@@ -48,9 +55,11 @@ struct event_create_operation : public base_operation
     */
    object_id_type event_group_id;
 
-   extensions_type   extensions;
+   extension< ext > extensions;
 
-   account_id_type fee_payer()const { return GRAPHENE_WITNESS_ACCOUNT; }
+   account_id_type fee_payer()const { 
+       return extensions.value.fee_paying_account ? *extensions.value.fee_paying_account : GRAPHENE_WITNESS_ACCOUNT; 
```

Comment:
I think you're falling into a trap here wrt authorization.

Before your change things are simple - GRAPHENE_WITNESS_ACCOUNT must approve the operation, which requires approval from majority of witness. When the evaluator is executed it knows that the witnesses have approved the operation and doesn't have to do any further checks.

Now you're separating things. The operation requires authority from the `fee_paying_account` if present, or from witnesses otherwise. This is fine as far as the fee payment is concerned. HOWEVER, the evaluator now has to check carefully who authorized the transaction. It does so here:
```
    FC_ASSERT(trx_state->_is_proposed_trx 
      || op.extensions.value.fee_paying_account ? is_manager( db(), op.event_group_id, *op.extensions.value.fee_paying_account ) : false,
      "trx is not proposed and fee_payer is not the manager of this object" );
```
This check is insufficient. It works for the witnesses and the manager, BUT it also works for everyone else *if they wrap the operation in a proposal*!

I recommend adding a unit test for this.

Reply:
If I understand your concern correctly. The potential issue is that a proposal could be created that uses a manager account and does then **not** validate the manager with `is_manager` because the `trx_state->_is_proposed_trx ` is `true` already.

So the fix could be to restrict this for either
* `trx_state->_is_proposed_trx && fee_paying_account = GRAPHENE_WITNESS_ACCOUNT`, or
* `op.extensions.value.fee_paying_account ? is_manager( db(), op.event_group_id, *op.extensions.value.fee_paying_account`

*(Note the additional check for fee_paying_account in the first option)*

Correct?

Reply:
Oh yeah thanks for pointing this out. This would be a high risk. But it is allready fixed with the hardfork protection code I added.